code/lib/cli/src/automigrate/fixes/initial-globals.test.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable no-underscore-dangle */
+import { vi, describe, expect, it } from 'vitest';
+import path from 'path';
+import * as fsExtra from 'fs-extra';
+import { initialGlobals } from './initial-globals';
+
+vi.mock('fs-extra', async () => import('../../../../../__mocks__/fs-extra'));
+
+const previewConfigPath = path.join('.storybook', 'preview.js');
+const check = async (previewContents: string) => {
+  vi.mocked<typeof import('../../../../../__mocks__/fs-extra')>(fsExtra as any).__setMockFiles({
+    [previewConfigPath]: previewContents,
+  });
+  return initialGlobals.check({
+    packageManager: {} as any,
+    configDir: '',
+    mainConfig: {} as any,
+    storybookVersion: '8.0',
+    previewConfigPath,
+  });
+};
+
+it('no globals setting', async () => {
+  await expect(check(`export default { tags: ['a', 'b']}`)).resolves.toBeFalsy();
+});
+
+it('initialGlobals setting', async () => {
+  await expect(check(`export default { initialGlobals: { a:  1 } }`)).resolves.toBeFalsy();
+});
+
+it('globals setting', async () => {
+  await expect(check(`export default { globals: { a:  1 } }`)).resolves.toBeTruthy();
+});
```

Comment:
What about non-default exports?

Reply:
This is already handled by `ConfigFile` unit tests

Reply:
I'm not sure I understand @shilman. If the user has `export const globals = {...}` what happens?

Reply:
`export const globals = { ... }` will get renamed to `export const initialGlobals = { .. }` (just hand-tested to verify).

The `ConfigFile` class takes care of three different config file syntaxes:
1. `export default { globals: { ... } }`
2. `export const globals = { ... }`
3. `module.exports = { globals: { } }` 

All of these cases are tested in the `ConfigFile` tests, so I didn't feel it was necessary to repeat those tests here. See https://github.com/storybookjs/storybook/blob/next/code/lib/csf-tools/src/ConfigFile.test.ts